### PR TITLE
Azure.ResourceManager.DeviceUpdate SDK update to the latest api-version

### PR DESCRIPTION
**Pull request by Sedols**
Update autorest.md with the new version. Regenerate SDK + update Test Session Recordings

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 # Release History
 
+## 1.0.0-beta.6 (Unreleased)
```
Comment (ArthurMa1978) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/CHANGELOG.md:3]: should be beta.5

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/autorest.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 azure-arm: true
 namespace: Azure.ResourceManager.DeviceUpdate
 require: https://github.com/Azure/azure-rest-api-specs/blob/32143b0f5f230ee2601e3c5d1990188666a5058d/specification/deviceupdate/resource-manager/readme.md
-tag: package-2020-03-01-preview
+tag: package-2022-04-01-preview
 clear-output-folder: true
```
Comment (ArthurMa1978) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/autorest.md:11]: please run `eng/script/Export-API.ps1 deviceupdate` to solve the ci failure

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+
+### Features Added
+
+### Breaking Changes
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/CHANGELOG.md:7]: Please add this section to `1.0.0-beta.5` and delete the changelog of `1.0.0-beta.6`.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Location.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.DeviceUpdate.Models
+{
+    /// <summary> The Location. </summary>
+    public partial class Location
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Location.cs:11]: Can you give this model a more specific name? `Location` is too general, and we also don't like single word class name.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Role.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.DeviceUpdate.Models
+{
+    /// <summary> Whether the location is primary or failover. </summary>
+    public readonly partial struct Role : IEquatable<Role>
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Role.cs:14]: Same as above, please rename this one.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/SKU.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.DeviceUpdate.Models
+{
+    /// <summary> Device Update Sku. </summary>
+    public readonly partial struct SKU : IEquatable<SKU>
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/SKU.cs:14]: Could we also rename this one? According to the .NET convention, we should use pascal case for acronyms over two letters. I think you can rename it to `Sku` and our generator will automatically change the name to `DeviceUpdateSku`.

**Review by Yao725 — COMMENTED** [review]
Left a few comments, you can use the directive in autorest.md to rename the model, thanks!

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Location.cs (hunk 4 above)
Comment (Sedols) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Location.cs:11]: This was automatically generated from the swagger. Are we allowed to edit these files?

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Role.cs (hunk 5 above)
Comment (Sedols) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Role.cs:14]: Same question, are we allowed to rename generated files?

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/SKU.cs (hunk 6 above)
Comment (Sedols) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/SKU.cs:14]: This was picked up from swagger https://github.com/Azure/azure-rest-api-specs/blob/c2d25c0f023edd1fea753a89d6409f3e7a9f8956/specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2022-04-01-preview/deviceupdate.json#L1335

              "x-ms-enum": {
                "name": "SKU",
                "modelAsString": true
              }

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/autorest.md (hunk 2 above)
Comment (Sedols) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/autorest.md:11]: I did but looks it is still failing with ChangeLog error, however it is present for the 1.0.0-beta.5 version, why is it failing still?

Error : No entry for version [1.0.0-beta.5] found in the ChangeLog [D:\a\_work\1\s\sdk\deviceupdate\Azure.ResourceManager.DeviceUpdate\CHANGELOG.md].

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/autorest.md [hunk 7]
```diff
@@ -64,4 +64,10 @@ directive:
           "type": "string"
         }
       }
+  - from: deviceupdate.json
+    where: $.definitions
+    transform: >
+      $.Location['x-ms-client-name'] = 'DeviceUpdateAccountLocationDetail';
+      $.Location.properties.role['x-ms-enum'].name = 'DeviceUpdateAccountLocationRole';
+      $.Account.properties.properties.properties.sku['x-ms-enum'].name = 'Sku';
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/autorest.md:72]: We don't edit these files, we use this autorest.md file to modify these names during the process of the generation. I renamed them using the configuration here and committed the change to your PR. You can take a look, if you're not ok with the names that I changed, feel free to rename them, just need to change the configuration.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Location.cs (hunk 4 above)
Comment (Yao725) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Location.cs]: Please see my latest comment, thanks!

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Role.cs (hunk 5 above)
Comment (Yao725) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/Role.cs]: Same as above.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/SKU.cs (hunk 6 above)
Comment (Yao725) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/SKU.cs]: Same as above.

**Review by Yao725 — APPROVED** [review]
LGTM

**Pull request merged**